### PR TITLE
Forked existing workflow to adapt for Azure CIP

### PR DESCRIPTION
**Pull request by DrizzlyOwl**

### CODE REVIEW

File: .github/workflows/deploy_to_environment_azure.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment (staging, production)"
+        required: true
+
+env:
+  DOCKER_IMAGE: a2bint-app
```
Comment (jordan-beard) [inline .github/workflows/deploy_to_environment_azure.yml:11]: Would it be alright to rename to conversions-app?
We want to move away from the name a2b internal eventually.

Comment (DrizzlyOwl) [reply .github/workflows/deploy_to_environment_azure.yml:11]: Yep that shouldn't be a problem. The only thing we'd need to update is the reference to the [image name in Terraform](https://github.com/DFE-Digital/a2b-internal/blob/main/terraform.gpaas-azure-migration/terraform.tfvars.example#L5)

File: .github/workflows/deploy_to_environment_azure.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Set environment variables
+        run: |
+          ENVIRONMENT=${{ github.event.inputs.environment }^^}
+
+          echo "Deploying to $ENVIRONMENT environment"
+
+          if [ "$ENVIRONMENT" == "STAGING" ]; then
```
Comment (jordan-beard) [inline .github/workflows/deploy_to_environment_azure.yml:88]: You might be uppercasing the environment name somewhere and I can't see it but wanted to make sure you were.

File: .github/workflows/deploy_to_environment_azure.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  #       with:
+  #         tag_name: ${{ env.RELEASE_TAGNAME }}
+  #         release_name: ${{ env.RELEASE_TAGNAME }}
+  #         draft: ${{ github.event.inputs.environment == 'staging' }}
+  #         prerelease: ${{ github.event.inputs.environment == 'staging' }}
+
+  cypress-tests:
+    needs: deploy-image
+    environment: staging
+    if: ${{ github.event.inputs.environment}} == staging
```
Comment (jordan-beard) [inline .github/workflows/deploy_to_environment_azure.yml:150]: You can see that down here the environment name check is slightly different. 
Not sure why its lower case, normally we enter it as 'Staging'

File: .github/workflows/deploy_to_environment_azure.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
+name: Deploy to environment (CIP)
```
Comment (jordan-beard) [inline .github/workflows/deploy_to_environment_azure.yml:1]: All makes sense to me. Thanks you.
As a nice to have enhancement do you know if there is a way to make it really obvious when viewing the summary of the run whether it was run for staging or production.
e.g. from this view

image

File: .github/workflows/deploy_to_environment_azure.yml (hunk 2 above)
Comment (DrizzlyOwl) [reply .github/workflows/deploy_to_environment_azure.yml:88]: [#L34](https://github.com/DFE-Digital/a2b-internal/blob/ops/github-action-workflow/.github/workflows/deploy_to_environment_azure.yml#L34) uses `^^` which is a `bash` way of Uppercasing a string. (untested in the context of a github action though)

File: .github/workflows/deploy_to_environment_azure.yml (hunk 4 above)
Comment (DrizzlyOwl) [reply .github/workflows/deploy_to_environment_azure.yml:1]: Github released "[Action Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/)" which seems like a perfect use for this. It supports conditions, contexts and variables and all rendered with GitHub flavoured markdown!

**Review by jordan-beard — APPROVED** [review]

**Pull request merged**